packages/rum/src/domain/record/record.spec.ts

```diff
@@ -311,6 +311,22 @@ describe('record', () => {
       expect(innerMutationData.isChecked).toBe(true)
     })
 
+    it('should record the change event inside a shadow root only once, regardless if the DOM is serialized multiple times', () => {
```

Comment:
This test is on `input` events, as it is easier to test. Mutations are asynchronous and since the original "Shadow Root Controller" was replaced by a new one, calling `flushMutations()` did not flush both of them.

Anyway, one test should be enough to avoid regression.